lib/model/rwfolder.go

```diff
@@ -1731,6 +1750,66 @@ func (f *sendReceiveFolder) IgnoresUpdated() {
 	f.IndexUpdated()
 }
 
+// deleteDir attempts to delete a directory. It checks for files/dirs inside
+// the directory and removes them if possible or returns an error if it fails
+func (f *sendReceiveFolder) deleteDir(dir string, ignores *ignore.Matcher, scanChan chan<- string) error {
+	files, _ := f.fs.DirNames(dir)
+
+	toBeDeleted := make([]string, 0)
```

Comment:
You are not going to delete more than there are files, and there ideally should not be any files, so perhaps it's worth using len(files) as a hint?